File: Programmers/[77886] 110 옮기기/woong-jae/77886.js
function solution(s) {
  const delete110 = x => {
      const stack = [];
      let count = 0;
      
      for(const char of x) {
          stack.push(char);
          if(stack.length < 3) continue;
          
          const top = stack.length - 1;
          if(stack[top] === "0" && stack[top - 1] === "1" && stack[top - 2] === "1"){
              for(let i = 0; i < 3; i++) stack.pop();
              count++;
          }
      }
      return [stack.join(""), count];
  }
  
  return s.map(x => {
      const [filtered_x, count] = delete110(x);
      
      const insertIndex = filtered_x.lastIndexOf("0") + 1;
      
      return filtered_x.slice(0, insertIndex) + "110".repeat(count) + filtered_x.slice(insertIndex);
  });
}

File: Programmers/[77886] 110 옮기기/Johoseong/77886.py
from collections import deque
def solution(s):
    answer = []

    for a in s:
        tmp = a
        stack = ''
        cnt = 0
        
        for t in tmp: # 110 없애기
            stack += t
            if stack[-3:] == '110':
                stack = stack[:-3]
                cnt += 1
        que = deque()
        stack = deque(list(stack))
        while stack: # 0 만나면 break (더 이상 앞에 110 넣어서 이득 없음)
            t = stack.pop()
            if t == '0':
                stack.append('0')
                break
            que.append(t)

        while cnt: # 110붙이고 나머지 1들 붙임
            cnt -= 1
            stack.append('1')
            stack.append('1')
            stack.append('0')
        stack += que
        answer.append(''.join(stack))
    return answer